### PR TITLE
Use general Write trait instead of ExportMethod.

### PR DESCRIPTION
**Pull request by luxrck**

### CODE REVIEW

**Comment by luxrck** [comment]
fix #201

File: sic_io/src/conversion.rs [hunk 1]
```diff
@@ -1,9 +1,8 @@
-use std::io::{self, Write};
-use std::path::Path;
+use std::io::Write;
 
 use sic_core::image;
 
-use crate::save::ExportMethod;
+// use crate::save::ExportMethod;
```
Comment (foresterre) [inline sic_io/src/conversion.rs]: Could you please remove the commented original code (1) :)

File: sic_io/src/conversion.rs [hunk 2]
```diff
@@ -45,16 +45,18 @@ impl<'a> ConversionWriter<'a> {
...
-                ConversionWriter::save_to_file(&export_buffer, output_format, v)
-            }
-            // None => write to stdout
-            ExportMethod::StdoutBytes => {
-                ConversionWriter::export_to_stdout(&export_buffer, output_format)
-            }
-        }
+        ConversionWriter::save_to(writer, &export_buffer, output_format)
+
+        // match export {
```
Comment (foresterre) [inline sic_io/src/conversion.rs]: Commented original code (2)

File: sic_io/src/conversion.rs [hunk 3]
```diff
@@ -91,38 +93,49 @@ impl<'a> ConversionWriter<'a> {
...
-    ) -> Result<(), String> {
-        let mut out = std::fs::File::create(path).map_err(|err| err.to_string())?;
-
-        buffer
-            .write_to(&mut out, format)
-            .map_err(|err| err.to_string())
-    }
-
-    fn export_to_stdout(
+    // fn save_to_file<P: AsRef<Path>>(
```
Comment (foresterre) [inline sic_io/src/conversion.rs]: Commented original code (3)

File: sic_io/src/conversion.rs [hunk 4]
```diff
@@ -143,7 +156,8 @@ mod tests {
         let conversion_processor = ConversionWriter::new(&buffer);
         conversion_processor
             .write(
-                ExportMethod::File(&output_path),
+                &mut File::create(&output_path)?,
+                // ExportMethod::File(&output_path),
```
Comment (foresterre) [inline sic_io/src/conversion.rs]: Commented original code (4)

File: sic_io/src/conversion.rs [hunk 5]
```diff
@@ -164,7 +179,8 @@ mod tests {
...
```
Comment (foresterre) [inline sic_io/src/conversion.rs]: Commented original code (5)

File: sic_io/src/conversion.rs [hunk 6]
```diff
@@ -188,7 +205,8 @@ mod tests {
...
```
Comment (foresterre) [inline sic_io/src/conversion.rs]: Commented original code (6)

File: sic_io/src/conversion.rs [hunk 7]
```diff
@@ -252,16 +271,17 @@ mod tests {
...
         expected_format: image::ImageFormat,
-    ) {
+    ) -> io::Result<()> {
         let our_output = &format!("header_match_conversion.{}", enc_format); // this is required because tests are run in parallel, and the creation, or deletion can collide.
         let output_path = setup_output_path(our_output);
 
         let buffer = image::open(setup_test_image(input)).expect("Can't open test file.");
         let conversion_processor = ConversionWriter::new(&buffer);
-        let method = ExportMethod::File(&output_path);
+        // let method = ExportMethod::File(&output_path);
```
Comment (foresterre) [inline sic_io/src/conversion.rs]: Commented original code (7)

File: sic_io/src/format.rs [hunk 8]
```diff
@@ -3,17 +3,17 @@ use std::path::Path;
 use sic_core::image;
 use std::error::Error;
 
-use crate::save::ExportMethod;
+// use crate::save::ExportMethod;
```
Comment (foresterre) [inline sic_io/src/format.rs]: Commented original code (8)

File: sic_io/src/format.rs [hunk 9]
```diff
@@ -3,17 +3,17 @@ use std::path::Path;
...
 
-const DEFAULT_PIPED_OUTPUT_FORMAT: image::ImageOutputFormat = image::ImageOutputFormat::BMP;
+// const DEFAULT_PIPED_OUTPUT_FORMAT: image::ImageOutputFormat = image::ImageOutputFormat::BMP;
```
Comment (foresterre) [inline sic_io/src/format.rs]: Commented original code (9)

File: sic_io/src/format.rs [hunk 10]
```diff
@@ -3,17 +3,17 @@ use std::path::Path;
...
 
-pub trait EncodingFormatByMethod {
-    /// Determine the encoding format based on the method of exporting.
-    fn by_method<P: AsRef<Path>>(
-        &self,
-        method: &ExportMethod<P>,
-    ) -> Result<image::ImageOutputFormat, Box<dyn Error>>;
-}
+// pub trait EncodingFormatByMethod {
```
Comment (foresterre) [inline sic_io/src/format.rs]: Commented original code (10)

File: sic_io/src/format.rs [hunk 11]
```diff
@@ -73,21 +73,21 @@ impl JPEGQuality {
...
-        &self,
-        method: &ExportMethod<P>,
-    ) -> Result<image::ImageOutputFormat, Box<dyn Error>> {
-        match method {
-            ExportMethod::StdoutBytes => Ok(DEFAULT_PIPED_OUTPUT_FORMAT),
-            ExportMethod::File(path) => self.by_extension(path),
-        }
-    }
-}
+// impl EncodingFormatByMethod for DetermineEncodingFormat {
```
Comment (foresterre) [inline sic_io/src/format.rs]: Commented original code (11)

File: sic_io/src/format.rs [hunk 12]
```diff
@@ -198,24 +198,24 @@ mod tests {
     }
 
     //
-    fn test_with_method_path(ext: &str, expected: &image::ImageOutputFormat) {
-        let path = format!("w_path.{}", ext);
-        let method = ExportMethod::File(path);
+    // fn test_with_method_path(ext: &str, expected: &image::ImageOutputFormat) {
```
Comment (foresterre) [inline sic_io/src/format.rs]: Commented original code (12)

File: sic_io/src/format.rs [hunk 13]
```diff
@@ -198,24 +198,24 @@ mod tests {
...
+    //     let path = format!("w_path.{}", ext);
+    //     let method = ExportMethod::File(path);
 
-        let format_determiner = setup_default_format_determiner();
-        let result = format_determiner.by_method(&method);
+    //     let format_determiner = setup_default_format_determiner();
```
Comment (foresterre) [inline sic_io/src/format.rs]: Commented original code (13)

File: sic_io/src/format.rs [hunk 14]
```diff
@@ -198,24 +198,24 @@ mod tests {
...
+    //     let result = format_determiner.by_method(&method);
 
-        assert_eq!(result.unwrap(), *expected);
-    }
+    //     assert_eq!(result.unwrap(), *expected);
```
Comment (foresterre) [inline sic_io/src/format.rs]: Commented original code (14)

File: sic_io/src/format.rs [hunk 15]
```diff
@@ -198,24 +198,24 @@ mod tests {
...
+    // }
 
-    #[test]
-    fn method_path_with_defaults() {
-        let zipped = INPUT_FORMATS.iter().zip(EXPECTED_VALUES.iter());
+    // #[test]
```
Comment (foresterre) [inline sic_io/src/format.rs]: Commented original code (15)

File: sic_io/src/format.rs [hunk 16]
```diff
@@ -198,24 +198,24 @@ mod tests {
...
+    // fn method_path_with_defaults() {
+    //     let zipped = INPUT_FORMATS.iter().zip(EXPECTED_VALUES.iter());
 
-        for (ext, exp) in zipped {
-            test_with_method_path(ext, exp);
-        }
-    }
+    //     for (ext, exp) in zipped {
```
Comment (foresterre) [inline sic_io/src/format.rs]: Commented original code (16)

File: sic_io/src/save.rs [hunk 17]
```diff
@@ -1,34 +1,36 @@
...
 pub struct ExportSettings {
     pub adjust_color_type: AutomaticColorTypeAdjustment,
 }
 
-#[derive(Debug)]
-pub enum ExportMethod<P: AsRef<Path>> {
-    File(P),
-    StdoutBytes,
-}
+// #[derive(Debug)]
```
Comment (foresterre) [inline sic_io/src/save.rs]: Commented original code (17)

File: sic_io/src/save.rs [hunk 18]
```diff
@@ -1,34 +1,36 @@
...
+// pub enum ExportMethod<P: AsRef<Path>> {
+//     File(P),
+//     StdoutBytes,
+// }
 
-// enum variants on type aliases are currently experimental, so we use a function here instead.
-pub fn use_stdout_bytes_as_export_method() -> ExportMethod<EmptyPath> {
-    ExportMethod::StdoutBytes
-}
+// // enum variants on type aliases are currently experimental, so we use a function here instead.
```
Comment (foresterre) [inline sic_io/src/save.rs]: Commented original code (18)

(I marked them all to have a quick overview to see them all marked as changed =); I'll try to do a proper review by tomorrow)

**Comment by foresterre** [comment]
Thanks!

**Pull request merged**